packages/wordcount/src/index.js

```diff
@@ -93,12 +93,13 @@ function matchCharacters( text, regex, settings ) {
 
 export function count( text, type, userSettings ) {
 	const settings = loadSettings( type, userSettings );
+	let returnData = 0;
 	if ( text ) {
 		let matchRegExp = settings[ type + 'RegExp' ];
-		if ( 'words' === settings.type ) {
-			return matchWords( text, matchRegExp, settings ).length;
-		} else {
-			return matchCharacters( text, matchRegExp, settings ).length;
+		const results = ( 'words' === settings.type ) ? matchWords( text, matchRegExp, settings ) : matchCharacters( text, matchRegExp, settings );
```

Comment:
This is quite a long line; I get why you're using a ternary here, but it might be nice to newline after the `:` 🤷‍♂️